--- node-v22.7.0/deps/v8/src/base/platform/yield-processor.h.orig	2024-08-21 23:12:04.000000000 +0200
+++ node-v22.7.0/deps/v8/src/base/platform/yield-processor.h	2024-08-25 14:35:10.922056543 +0200
@@ -41,7 +41,7 @@
 #if defined(V8_HOST_ARCH_IA32) || defined(V8_HOST_ARCH_X64)
 #define YIELD_PROCESSOR __asm__ __volatile__("pause")
 #elif defined(V8_HOST_ARCH_ARM64) || \
-    (defined(V8_HOST_ARCH_ARM) && __ARM_ARCH >= 6)
+    (defined(V8_HOST_ARCH_ARM) && __ARM_ARCH >= 7)
 #define YIELD_PROCESSOR __asm__ __volatile__("yield")
 #elif defined(V8_HOST_ARCH_MIPS64EL) && __mips_isa_rev >= 2
 // Don't bother doing using .word here since r2 is the lowest supported mips64
