--- ../node-v0.4.11.original/lib/module.js	2011-08-17 18:18:10.000000000 -0700
+++ ./lib/module.js	2011-08-22 14:37:06.590597180 -0700
@@ -422,7 +422,7 @@
 };
 
 Module._initPaths = function() {
-  var paths = [path.resolve(process.execPath, '..', '..', 'lib', 'node')];
+  var paths = ["/usr/lib64/node", "/usr/lib/node", "/usr/lib/node_modules"];
 
   if (process.env['HOME']) {
     paths.unshift(path.resolve(process.env['HOME'], '.node_libraries'));
--- node-v0.6.8/./tools/node-waf~	2012-01-20 04:15:27.000000000 +0200
+++ node-v0.6.8/./tools/node-waf	2012-01-26 17:23:09.336122520 +0200
@@ -5,7 +5,7 @@
 join = os.path.join
 bindir = os.path.dirname(os.path.realpath(__file__))
 prefix = join(bindir, "..")
-wafdir = join(prefix, "lib", "node")
+wafdir = join(prefix, "lib64", "node")
 
 w = join(wafdir, 'wafadmin')
 t = join(w, 'Tools')
