diff -ur node-v0.4.0-o//wscript node-v0.4.0/wscript
--- node-v0.4.0-o//wscript	2011-02-10 03:40:11.000000000 -0700
+++ node-v0.4.0/wscript	2011-02-19 00:47:27.000000000 -0700
@@ -384,7 +384,6 @@
   # LFS
   conf.env.append_value('CPPFLAGS',  '-D_LARGEFILE_SOURCE')
   conf.env.append_value('CPPFLAGS',  '-D_FILE_OFFSET_BITS=64')
-  conf.env.append_value('CPPFLAGS',  '-DEV_MULTIPLICITY=0')
 
   # Makes select on windows support more than 64 FDs
   if sys.platform.startswith("win32"):
Only in node-v0.4.0: wscript~
