--- node-v12.14.1/lib/internal/modules/cjs/loader.js~	2020-01-07 13:16:01.000000000 +0100
+++ node-v12.14.1/lib/internal/modules/cjs/loader.js	2020-01-25 23:09:02.305347922 +0100
@@ -1095,7 +1095,7 @@ Module._initPaths = function() {
     path.resolve(process.execPath, '..') :
     path.resolve(process.execPath, '..', '..');
 
-  let paths = [path.resolve(prefixDir, 'lib', 'node')];
+  let paths = ["/usr/lib/node", "/usr/lib/node_modules", path.resolve(prefixDir, 'lib', 'node')];
 
   if (homeDir) {
     paths.unshift(path.resolve(homeDir, '.node_libraries'));
