--- node-v10.16.0/lib/internal/modules/cjs/loader.js~	2019-05-28 23:32:23.000000000 +0200
+++ node-v10.16.0/lib/internal/modules/cjs/loader.js	2019-06-03 09:11:35.662170541 +0200
@@ -861,7 +861,7 @@ Module._initPaths = function() {
   } else {
     prefixDir = path.resolve(process.execPath, '..', '..');
   }
-  var paths = [path.resolve(prefixDir, 'lib', 'node')];
+  var paths = ["/usr/lib/node", "/usr/lib/node_modules", path.resolve(prefixDir, 'lib', 'node')];
 
   if (homeDir) {
     paths.unshift(path.resolve(homeDir, '.node_libraries'));
