diff -ur node-v0.1.92-o//wscript node-v0.1.92/wscript
--- node-v0.1.92-o//wscript	2010-04-24 01:37:14.000000000 +0200
+++ node-v0.1.92/wscript	2010-04-24 04:53:25.000000000 +0200
@@ -436,7 +436,7 @@
     """
     node.add_objects = 'eio evcom http_parser coupling'
     node.uselib_local = 'eio'
-    node.uselib = 'RT EV OPENSSL GNUTLS GPGERROR UDNS V8 EXECINFO DL KVM SOCKET NSL'
+    node.uselib = 'RT CARES EV OPENSSL GNUTLS GPGERROR UDNS V8 EXECINFO DL KVM SOCKET NSL'
 
   node.install_path = '${PREFIX}/lib'
   node.install_path = '${PREFIX}/bin'
